src/server/redis_connection.h

```diff
@@ -90,6 +90,9 @@ class Connection {
   uint32_t GetPort() { return port_; }
   void SetListeningPort(int port) { listening_port_ = port; }
   int GetListeningPort() { return listening_port_; }
+  void SetAnnounceIP(std::string ip) { announce_ip_ = std::move(ip); }
+  std::string GetAnnounceIP() { return announce_ip_; }
+  std::string GetAnnounceIPOrIP() { return !announce_ip_.empty() ? announce_ip_ : ip_; }
```

Comment:
We can do this directly in the `GetIP()` function. If you want a different name, delete `GetIP()` which not used anywhere else.

Reply:
GetAnnounceIP is only meaning to slave connections and others should also use GetIP, so we need to keep the GetIP function.

Reply:
There is no problem keeping this, individuals prefer when to use and when to add.

Reply:
Yes, it's using indeed now.